### PR TITLE
BUG: Prevent abline from updating non-existent line

### PR DESCRIPTION
**Pull request by bashtage**
Prevent abline form updating non-existant line
Fixes a bug that appears in robust_models_1 notebook

### CODE REVIEW

File: statsmodels/graphics/regressionplots.py [hunk 1]
```diff
@@ -694,23 +694,34 @@ def abline_plot(intercept=None, slope=None, horiz=None, vert=None,
...
             ax.figure.canvas.draw()
-    #TODO: how to intercept something like a margins call and adjust?
 
+    # TODO: how to intercept something like a margins call and adjust?
     line = ABLine2D(x, data_y, **kwargs)
     ax.add_line(line)
-    ax.callbacks.connect('xlim_changed', line.update_datalim)
-    ax.callbacks.connect('ylim_changed', line.update_datalim)
+    line.id_xlim_callback = ax.callbacks.connect('xlim_changed', line.update_datalim)
+    line.id_ylim_callback = ax.callbacks.connect('ylim_changed', line.update_datalim)
```
Comment (josef-pkt) [inline statsmodels/graphics/regressionplots.py]: I just realized that the use of `callbacks` is not new. So there is no need to worry about impact of callbacks anymore.
(I'm not sure what the use of callbacks assumes about the matplotlib backend.)

**Comment by josef-pkt** [comment]
This contains merge conflict artifacts `<<<<<<< HEAD`

**Comment by bashtage** [comment]
Fixed that.

**Comment by bashtage** [comment]
@josef-pkt I think this small PR is ready

**Comment by josef-pkt** [comment]
merging, (I kind of understand this)

@bashtage  Thanks

**Pull request merged**